src/parser/ion_1_0/text/mod.rs

```diff
@@ -0,0 +1,2614 @@
+}
+
+///
+/// Encoding Section: Ion TextTimestamp
+///
+
+/// Encoding: fragment TIMESTAMP
+/// TIMESTAMP
+///     : DATE ('T' TIME?)?
+///     | YEAR '-' MONTH 'T'
+///     | YEAR 'T'
+///     ;
+fn take_timestamp(i: &str) -> IonResult<&str, ion::Timestamp> {
+    let (i, timestamp) = map(
+        alt((
+            map_res(
+                pair(take_date, opt(preceded(one_of("tT"), opt(take_time)))),
+                |((year, month, day), maybe_time)| match ion::Date::day(year, month, day) {
+                    Ok(date) => {
+                        if let Some(time) = maybe_time {
+                            Ok(ion::TextTimestamp::new(date, time))
+                        } else {
+                            Ok(ion::TextTimestamp::new(date, None))
+                        }
+                    }
+                    Err(e) => Err(e),
+                },
+            ),
+            // Timestamps which consist of only a year or year and month must be terminated by t or T.
+            map(
+                terminated(
+                    separated_pair(take_year, char('-'), take_month),
+                    one_of("tT"),
+                ),
+                |(year, month)| ion::TextTimestamp::new(ion::Date::Month { year, month }, None),
+            ),
+            map(terminated(take_year, one_of("tT")), |year| {
+                ion::TextTimestamp::new(ion::Date::Year { year }, None)
+            }),
+        )),
+        ion::Timestamp::Text,
+    )(i)?;
+
+    Ok((i, timestamp))
+}
+
+/// fragment
+/// DATE
+///     : YEAR '-' MONTH '-' DAY
+///     ;
+fn take_date(i: &str) -> IonResult<&str, (i32, u8, u8)> {
+    let (i, (y, (m, d))) = pair(
+        take_year,
+        pair(
+            preceded(char('-'), take_month),
+            preceded(char('-'), take_day),
+        ),
+    )(i)?;
+
+    Ok((i, (y, m, d)))
+}
+
+/// fragment
+/// YEAR
+///     : '000'                     [1-9]
+///     | '00'            [1-9]     DEC_DIGIT
+///     | '0'   [1-9]     DEC_DIGIT DEC_DIGIT
+///     | [1-9] DEC_DIGIT DEC_DIGIT DEC_DIGIT
+///     ;
+fn take_year(i: &str) -> IonResult<&str, i32> {
+    map(
+        alt((
+            recognize(pair(tag("000"), one_of("123456789"))),
+            recognize(tuple((
+                tag("00"),
+                one_of("123456789"),
+                one_if(is_dec_digit),
+            ))),
+            recognize(tuple((
+                tag("0"),
+                one_of("123456789"),
+                take_while_m_n(2, 2, is_dec_digit),
+            ))),
+            recognize(tuple((
+                one_of("123456789"),
+                take_while_m_n(3, 3, is_dec_digit),
+            ))),
+        )),
+        |s: &str| {
+            s.parse::<i32>()
+                .expect("the parser ensures it will be within range")
+        },
+    )(i)
+}
+
+/// fragment
+/// MONTH
+///     : '0' [1-9]
+///     | '1' [0-2]
+///     ;
+fn take_month(i: &str) -> IonResult<&str, u8> {
```

Comment:
Does `time` have an enum we can use for the response type?

Reply:
It does not. The only enum it provides is `Weekday`, which is interesting.